src/accessors/IIterator.ts

```diff
@@ -1,6 +1,6 @@
-import { IIteratorResult } from './IIteratorResult';
-
 export interface IIterator<T> {
-    next(skip: number): IIteratorResult<T>;
+    next(packageSize?: number): Array<T>;
     [Symbol.iterator](): IIterator<T>;
+
+    hasMoreElements(): boolean;
```

Comment:
Check my comment above regarding the iterator protocol, but I agree having this method would be good. Maybe let's change the name to just `hasMore()`?